### ISSUE TITLE
[no-render-in-lifecycle] false positive

### ISSUE THREAD

**Issue by Maxim-Mazurok** [testing-library/eslint-plugin-testing-library#1000]
### Have you read the Troubleshooting section?

Yes

### Plugin version

v6.1.2

### ESLint version

v8.57.0

### Node.js version

v20.17.0

### package manager and version

npm 10.8.2

### Operating system

Ubuntu 24

### Bug description

[no-render-in-lifecycle](https://github.com/testing-library/eslint-plugin-testing-library/blob/main/docs/rules/no-render-in-lifecycle.md) gives false posivite when calling a function that starts with `render`:
```ts
beforeEach(() => {
  setActivePinia(createTestingPinia());
  renderStoreInsideSetup(useLoggedInUserStore); // This throws false-positive - unexpected
  useComplianceStore().landingPageNavigationTriggered = true;
});
```
Likely due to regex issue?

### Steps to reproduce

Lint code from above

### Error output/screenshots

```
Forbidden usage of `render` within testing framework `beforeEach` setupeslint[testing-library/no-render-in-lifecycle](https://github.com/testing-library/eslint-plugin-testing-library/tree/main/docs/rules/no-render-in-lifecycle.md)
```

### ESLint configuration

```
...

...compat
    .extends("plugin:jest-formatting/recommended", "plugin:vitest/all", "plugin:testing-library/vue")
    .map((config) => ({
      ...config,
      files: ["src/**/__tests__/*", "**/*.spec.ts", "**/*.spec.tsx", "**/*.skip.ts"],
    })),

...
```

### Rule(s) affected

testing-library/no-render-in-lifecycle

### Do you want to submit a pull request to fix this bug?

No

**Comment by Maxim-Mazurok** [comment]
Actually after reading more about the rule origin, maybe it's not a false positive... We don't actually render the component, we just use this to test pinia stores that use vue-query which relies on being inside of setup(), so we create a dummy component for these tests.

I guess, according to the [article](https://kentcdodds.com/blog/avoid-nesting-when-youre-testing) we should create a setup function instead and manually call it in each test..., is that the idea? Or is this rule meant to only apply to actually rendering the component?

Originally I said that it's a false-positive because examples only mention `render()` name, not `renderSomethingCustom()`

**Comment by Belco90** [comment]
Hi @Maxim-Mazurok.

The original idea behind this rule is to avoid rendering the tested component in the `beforeEach` indeed. However, your `renderStoreInsideSetup` is reported as a rendering method because of [the Aggressive Reporting](https://github.com/testing-library/eslint-plugin-testing-library#aggressive-reporting).

If you read [the custom renders behavior doc](https://github.com/testing-library/eslint-plugin-testing-library/blob/main/docs/migration-guides/v4.md#renders), you'll see the plugin will assume any function containing "render" in its name will be considered as a proper Testing Library rendering method, hence the reported error in your code.

You can customize the setting for this behavior, ignore the corresponding line reported if your `renderStoreInsideSetup` is not actually rendering components, or just rename your util to something else to indicate it doesn't actually render components.

**Comment by Maxim-Mazurok** [comment]
Makes perfect sense, thanks for linking to the docs, appreciate that this is configurable as well!
In my case, I just renamed the util function to not include "render".
It technically renders a dummy component in order to be able to use store, but it's not rendering an actual component, so I think it's fine...
Thanks, closing for now!
